syz-manager/manager.go

```diff
@@ -545,6 +554,22 @@ func (mgr *Manager) loadProg(data []byte, minimized, smashed bool) bool {
 	return true
 }
 
+func programLeftover(target *prog.Target, enabled map[*prog.Syscall]bool, data []byte) []byte {
```

Comment:
It feels it should be part of checkProgram, it already deserializes, checks for errors, and checks for disabled calls. Can it be a flag of checkProgram?

Reply:
Yes, it feels so and I thought about it, but it seemed to me that it's just not worth it.
* `checkProgram` is invoked in 4 different places, but only one needs to cut out disabled calls.
*  `checkProgram`  would become too complicated - 4 input parameters and 3 output parameters.
* At the same time, prog parsing is not our bottleneck. Moreover, `programLeftover` will only be called after we disabled some syscalls, the next time syz-manager is started there's no need to invoke it as there are no more disabled progs.